config/main.go

```diff
@@ -37,3 +44,93 @@ var (
 	// if RequirePass is set to an empty string, no authentication is required
 	RequirePass string = constants.EmptyStr
 )
+
+func init() {
+	args := os.Args[1:]
+
+	if len(args) == 1 && args[0] == constants.InitConfig {
```

Comment:
Would suggest breaking down the function into creation and handling of config files.
Also if possible try to reduce nesting by eliminating code paths earlier in code e.g:
```
if len(args) != 1 || args[0] != InitConfig {
    return
}
```